src/lib.rs

```diff
@@ -2972,6 +3046,162 @@ impl GooseAttack {
+                aggregate_response_time_counter += request.total_response_time;
+                aggregate_response_time_minimum = metrics::update_min_time(
+                    aggregate_response_time_minimum,
+                    request.min_response_time,
+                );
+                aggregate_response_time_maximum = metrics::update_max_time(
+                    aggregate_response_time_maximum,
+                    request.max_response_time,
+                );
+                aggregate_response_times =
+                    metrics::merge_times(aggregate_response_times, request.response_times.clone());
+            }
+
+            // Prepare aggregate per-request metrics.
+            let (aggregate_requests_per_second, aggregate_failures_per_second) =
+                metrics::per_second_calculations(
+                    self.metrics.duration,
+                    aggregate_total_count,
+                    aggregate_fail_count,
+                );
+            request_metrics.push(report::RequestMetric {
+                method: "".to_string(),
+                name: "Aggregated".to_string(),
+                number_of_requests: aggregate_total_count,
+                number_of_failures: aggregate_fail_count,
+                response_time_average: format!(
+                    "{:.2}",
+                    aggregate_response_time_counter as f32 / aggregate_total_count as f32
+                ),
+                response_time_minimum: aggregate_response_time_minimum,
+                response_time_maximum: aggregate_response_time_maximum,
+                requests_per_second: format!("{:.2}", aggregate_requests_per_second),
+                failures_per_second: format!("{:.2}", aggregate_failures_per_second),
+            });
+
+            // Prepare aggregate per-response metrics.
+            response_metrics.push(report::get_response_metric(
+                "",
+                "Aggregated",
+                &aggregate_response_times,
+                aggregate_total_count,
+                aggregate_response_time_minimum,
+                aggregate_response_time_maximum,
+            ));
+
+            // Compile the request metrics template.
+            let mut request_template = Vec::new();
+            for metric in request_metrics {
+                request_template.push(report::request_metrics_template(metric));
+            }
+
+            // Compile the response metrics template.
+            let mut response_template = Vec::new();
+            for metric in response_metrics {
+                response_template.push(report::response_metrics_template(metric));
+            }
+
+            // Compile the report template.
+            let report = report::build_report(
+                &start_time,
+                &end_time,
+                &host,
+                &request_template.join("\n"),
+                &response_template.join("\n"),
+            );
```

Comment:
@LionsAd And here's the "outer" template -- this builds the actual HTML report, and inserts the inner templates we built earlier.